syntax = "proto3";

package calculus;

service VectorCalculator {
  rpc CrossProduct (VectorRequest) returns (Vector) {}
}

message Scalar {
  double value = 1;
}

message Vector {
  double x = 1;
  double y = 2;
  double z = 3;
}

message VectorRequest {
  Vector first = 1;
  Vector second = 2;
  Vector thrird = 3;
}
